### ISSUE TITLE
attrs: wrong type inferred from Factory attribute (using auto_attribs)

### ISSUE THREAD

**Issue by belm0** [pylint-dev/pylint#4341]
### Steps to reproduce

attrs allows this form for factory, but pylint doesn't understand it:
```python3
import attr
from functools import partial
import numpy as np

@attr.attrs(auto_attribs=True)
class Foo:
    position: np.ndarray = attr.Factory(partial(np.zeros, 3))
    # position: np.ndarray = attr.attrib(default=attr.Factory(partial(np.zeros, 3)))  # this works

    def bar(self):
        return -self.position
```

### Current behavior
```
 E1130: bad operand type for unary -: Factory (invalid-unary-operand-type)
```

### Expected behavior
no error

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.7.2
astroid 2.5.1
Python 3.7.7 (default, Oct  5 2020, 09:18:33)
[Clang 9.0.0 (clang-900.0.39.2)]
```

Additional dependencies:
```
attrs=20.2.0
```

**Comment by hippo91** [comment]
@belm0 thanks for your report. Would it be possible to have smaller reproducer? For example without `numpy` dependency?
It would be clearer if this false positive arises from our handling of `attr` or `numpy`.

**Comment by belm0** [comment]
confirmed it's not related to numpy:

```python
import attr

@attr.attrs(auto_attribs=True)
class Foo:
    position: int = attr.Factory(lambda: 3)
    #position: int = attr.attrib(default=attr.Factory(lambda: 3))  # this works

    def bar(self):
        return -self.position
```

```
E1130: bad operand type for unary -: Factory (invalid-unary-operand-type)
```

**Comment by hippo91** [comment]
@belm0 thanks for this update!
